packages/manager/src/features/ErrorBoundary/ErrorComponent.tsx

```diff
@@ -104,3 +106,23 @@ export const ErrorComponent = ({
     </Box>
   );
 };
+
+const normalizeError = (error: unknown): Error => {
+  if (error instanceof Error) {
+    return error;
+  }
+
+  if (typeof error === 'string') {
+    return new Error(error);
+  }
+
+  if (Array.isArray(error) && error.length === 1 && error[0]?.reason) {
+    return new Error(error[0].reason);
+  }
+
+  if (typeof error === 'function' || error === undefined) {
+    return new Error('Unknown error');
+  }
+
+  return new Error(JSON.stringify(error));
```

Comment:
Suggesting some protection in case `error` has circular references: 
```suggestion
  try {
    return new Error(JSON.stringify(error));
  } catch {
    return new Error('Unserializable error');
  }
```

Reply:
good call 👍